### PR TITLE
bugfix: disable `ZipkinAutoConfiguration`

### LINKED ISSUE #1630: spring-cloud-gcp-starter-trace 4.1.1 WARN logs after fix #1568
After fix #1568 for issue #1554, in the spring-cloud-gcp-trace-sample (version 4.1.1) app logs, after making a first GET request to http://localhost:8080, I see the following WARN logs:
```
WARN  z.r.AsyncReporter$BoundedAsyncReporter   : Spans were dropped due to exceptions. All subsequent errors will be logged at FINE level.
WARN  z.r.AsyncReporter$BoundedAsyncReporter   : Dropped 8 spans due to AssertionError(com.google.protobuf.InvalidProtocolBufferException: Protocol message end-group tag did not match expected tag.)

java.lang.AssertionError: com.google.protobuf.InvalidProtocolBufferException: Protocol message end-group tag did not match expected tag.
	at zipkin2.reporter.stackdriver.StackdriverSender.parseTraceIdPrefixedSpan(StackdriverSender.java:220)
	at zipkin2.reporter.stackdriver.StackdriverSender.sendSpans(StackdriverSender.java:168)
	at zipkin2.reporter.AsyncReporter$BoundedAsyncReporter.flush(AsyncReporter.java:299)
	at zipkin2.reporter.AsyncReporter$Flusher.run(AsyncReporter.java:378)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.google.protobuf.InvalidProtocolBufferException: Protocol message end-group tag did not match expected tag.
	at com.google.protobuf.InvalidProtocolBufferException.invalidEndTag(InvalidProtocolBufferException.java:137)
	at com.google.protobuf.CodedInputStream$ArrayDecoder.checkLastTagWas(CodedInputStream.java:641)
	at com.google.protobuf.CodedInputStream$ArrayDecoder.readMessage(CodedInputStream.java:873)
	at com.google.devtools.cloudtrace.v2.Span$Builder.mergeFrom(Span.java:10357)
	at com.google.devtools.cloudtrace.v2.Span$Builder.mergeFrom(Span.java:10043)
	at com.google.protobuf.AbstractMessage$Builder.mergeFrom(AbstractMessage.java:418)
	at com.google.protobuf.AbstractMessage$Builder.mergeFrom(AbstractMessage.java:317)
	at com.google.protobuf.AbstractMessageLite$Builder.mergeFrom(AbstractMessageLite.java:222)
	at com.google.protobuf.AbstractMessage$Builder.mergeFrom(AbstractMessage.java:535)
	at zipkin2.reporter.stackdriver.StackdriverSender.parseTraceIdPrefixedSpan(StackdriverSender.java:218)
	... 4 common frames omitted
```

If I add back the exclusion of ZipkinAutoConfiguration,
```
/** Sample spring boot application. */
@AutoConfiguration
@SpringBootApplication(exclude = ZipkinAutoConfiguration.class)
public class Application implements WebMvcConfigurer {
```
I don't see the WARN logs anymore.

Expected would be to not see the warnings when the exclude is missing.

### PR DESCRIPTION
**Pull request by JoeWang1127**
Fix #1630.

### CODE REVIEW

File: spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/TraceAutoConfigurationFilter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  private static final Set<String> SHOULD_SKIP = Set.of(
+      "org.springframework.boot.actuate.autoconfigure.tracing.zipkin.ZipkinAutoConfiguration");
+
+  @Override
+  public boolean[] match(String[] autoConfigurationClasses,
+      AutoConfigurationMetadata autoConfigurationMetadata) {
+    boolean[] matches = new boolean[autoConfigurationClasses.length];
+
+    for (int i = 0; i < autoConfigurationClasses.length; i++) {
+      matches[i] = autoConfigurationClasses[i] == null
```
Comment (JoeWang1127) [inline spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/TraceAutoConfigurationFilter.java:34]: reference [code](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/condition/FilteringSpringBootCondition.java#L52)

**Comment by JoeWang1127** [comment]
The warning is gone and no dropped spans
```
13:57:44.689 [main] INFO  c.e.TraceSampleApplicationIntegrationTests - Started TraceSampleApplicationIntegrationTests in 2.345 seconds (process running for 10.223)
13:57:44.966 [http-nio-auto-1-exec-1] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring DispatcherServlet 'dispatcherServlet'
13:57:44.967 [http-nio-auto-1-exec-1] INFO  o.s.web.servlet.DispatcherServlet - Initializing Servlet 'dispatcherServlet'
13:57:44.968 [http-nio-auto-1-exec-1] INFO  o.s.web.servlet.DispatcherServlet - Completed initialization in 1 ms
13:57:44.997 [http-nio-auto-1-exec-1] INFO  com.example.WorkService - starting busy work
13:57:45.394 [http-nio-auto-1-exec-2] INFO  com.example.ExampleController - meeting took 386ms
13:57:46.095 [http-nio-auto-1-exec-3] INFO  com.example.ExampleController - meeting took 651ms
13:57:46.469 [http-nio-auto-1-exec-4] INFO  com.example.ExampleController - meeting took 345ms
13:57:46.472 [http-nio-auto-1-exec-1] INFO  com.example.WorkService - finished busy work
13:57:48.115 [global-gcp-pubsub-subscriber1] INFO  com.example.Application - Message arrived! Payload: All work is done via PubSubTemplate.
13:57:48.237 [global-gcp-pubsub-subscriber1] INFO  com.example.Application - Message arrived! Payload: All work is done via SI.
13:57:51.353 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - Found trace! 1328b698a4df4eab8cf4acdb463b29e3 with 16 spans ([http get /, visit-meet-endpoint, http get, http get /meet, http get, http get /meet, http get, http get /meet, send-message-spring-integration, publish, send-message-pub-sub-template, publish, next-message, on-message, next-message, on-message]).
13:57:52.361 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - 
========================= [START OF LOG CONTENTS] =========================
[meeting took 386ms, meeting took 651ms, meeting took 345ms, finished busy work]
========================= [END OF LOG CONTENTS]   =========================

13:57:54.510 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - Found trace! 1328b698a4df4eab8cf4acdb463b29e3 with 16 spans ([http get /, visit-meet-endpoint, http get, http get /meet, http get, http get /meet, http get, http get /meet, send-message-spring-integration, publish, send-message-pub-sub-template, publish, next-message, on-message, next-message, on-message]).
13:57:57.039 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - 
========================= [START OF LOG CONTENTS] =========================
[meeting took 386ms, meeting took 651ms, meeting took 345ms, finished busy work]
========================= [END OF LOG CONTENTS]   =========================

13:57:59.195 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - Found trace! 1328b698a4df4eab8cf4acdb463b29e3 with 16 spans ([http get /, visit-meet-endpoint, http get, http get /meet, http get, http get /meet, http get, http get /meet, send-message-spring-integration, publish, send-message-pub-sub-template, publish, next-message, on-message, next-message, on-message]).
13:58:00.064 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - 
========================= [START OF LOG CONTENTS] =========================
[meeting took 386ms, meeting took 651ms, meeting took 345ms, finished busy work]
========================= [END OF LOG CONTENTS]   =========================

13:58:02.225 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - Found trace! 1328b698a4df4eab8cf4acdb463b29e3 with 16 spans ([http get /, visit-meet-endpoint, http get, http get /meet, http get, http get /meet, http get, http get /meet, send-message-spring-integration, publish, send-message-pub-sub-template, publish, next-message, on-message, next-message, on-message]).
13:58:03.138 [awaitility-thread] INFO  c.e.TraceSampleApplicationIntegrationTests - 
========================= [START OF LOG CONTENTS] =========================
[starting busy work, meeting took 386ms, meeting took 651ms, meeting took 345ms, finished busy work, Message arrived! Payload: All work is done via PubSubTemplate., Message arrived! Payload: All work is done via SI.]
========================= [END OF LOG CONTENTS]   =========================
```

File: spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/TraceAutoConfigurationFilter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.trace;
+
+import java.util.Set;
+import org.springframework.boot.autoconfigure.AutoConfigurationImportFilter;
+import org.springframework.boot.autoconfigure.AutoConfigurationMetadata;
+
+public class TraceAutoConfigurationFilter implements AutoConfigurationImportFilter {
```
Comment (meltsufin) [inline spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/TraceAutoConfigurationFilter.java:23]: Please add a javadoc comment here describing the purpose.

**Review by meltsufin — APPROVED** [review]
LGTM! Thanks!

**Review by meltsufin — APPROVED** [review]

**Pull request merged**